### ISSUE TITLE
junos: connection succeed only after 3/4 runs

### ISSUE THREAD

**Issue by alvarezluciano** [ansible/ansible#33678]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- ncclient
- connection plugin

##### ANSIBLE VERSION

```
ansible 2.4.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]

```

##### CONFIGURATION

```
DEFAULT_LOG_PATH(/etc/ansible/ansible.cfg) = /var/log/ansible.log
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 20
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
PARAMIKO_HOST_KEY_AUTO_ADD(/etc/ansible/ansible.cfg) = True
PARAMIKO_LOOK_FOR_KEYS(/etc/ansible/ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/etc/ansible/ansible.cfg) = 90
PERSISTENT_CONNECT_RETRY_TIMEOUT(/etc/ansible/ansible.cfg) = 90
PERSISTENT_CONNECT_TIMEOUT(/etc/ansible/ansible.cfg) = 300
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
```
##### OS / ENVIRONMENT

CentOS 7

##### SUMMARY

Trying to run some Junos commands, and it only succeds after running 3/4/5 times (random).
Enable netconf  task (runs before the task that is failing) runs successfully ALWAYS.

##### ACTUAL RESULTS

```

ansible.log output:
2017-12-06 07:41:52,966 p=5954 u=root |  control socket path is /root/.ansible/pc/acf31d59a2
2017-12-06 07:41:52,966 p=5954 u=root |  current working directory is /etc/ansible
2017-12-06 07:41:52,966 p=5954 u=root |  using connection plugin netconf
2017-12-06 07:41:53,035 p=5954 u=root |  network_os is set to junos
2017-12-06 07:41:53,038 p=5954 u=root |  ssh connection done, stating ncclient
2017-12-06 07:41:53,041 p=5954 u=root |  failed to create control socket for host 10.143.57.4
2017-12-06 07:41:53,042 p=5954 u=root |  Traceback (most recent call last):
  File "/usr/bin/ansible-connection", line 315, in main
    server = Server(socket_path, pc)
  File "/usr/bin/ansible-connection", line 112, in __init__
    self.connection._connect()
  File "/usr/lib/python2.7/site-packages/ansible/plugins/connection/netconf.py", line 158, in _connect
    ssh_config=ssh_config
  File "/usr/lib/python2.7/site-packages/ncclient/manager.py", line 154, in connect
    return connect_ssh(*args, **kwds)
  File "/usr/lib/python2.7/site-packages/ncclient/manager.py", line 119, in connect_ssh
    session.connect(*args, **kwds)
  File "/usr/lib/python2.7/site-packages/ncclient/transport/ssh.py", line 383, in connect
    raise SSHError("Could not open socket to %s:%s" % (host, port))
SSHError: Could not open socket to 10.143.57.4:830

2017-12-06 07:43:23,495 p=5073 u=root |  fatal: [USX2ZNWXS13-(QFX-3500-1C4C,4D)]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "unable to open shell. Please see: https://docs.ansible.com/ansible/network_debug_troubleshooting.html#unable-to-open-shell"
}
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/plugins/connection](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
Hi, thanks for the bug report
1) Does `junos_netconf` always work
2) Does `junos_netconf` always report OK(no change) or CHANGED?
We are wondering if there is an issue when netconf is getting enabled and it's taking a few seconds before it's up and running, though by that point we've already failed the following task.

**Comment by ganeshrn** [comment]
@alvarezluciano To confirm on what @gundalow mentioned in above comment can you please add a delay after the first task that enables netconf on the remote host and before the second task that uses netconf transport to connect to the remote device. 

Delay can be added by using below task
```
- name: wait for netconf server to come up
  pause:
    seconds: 10
```
This might be a latency issue with Netconf server taking time to come up.

needs_info

**Comment by alvarezluciano** [comment]
@gundalow @ganeshrn 
Thanks for the replies! I came up with the same idea, I tried to wait a certain time, but I did it after closing again the netconf port... But now I included the delay task (30 seconds to be sure), and it seems to be working on MOST of them. Actually, at the time only one failed (same error), and even with 2min delay.
Junos_netconf always works, and reports correctly (ok/changed depending the actual state).

I will report final results next week!

**Comment by alvarezluciano** [comment]
Updating the issue...
Some of them are still failing with same error output. I'm using a 1min delay.
Any other suggestions/ideas?
Thanks in advance!

**Comment by ganeshrn** [comment]
@alvarezluciano Can you please retest on `stable-2.5` branch.
Also please share the playbook you are running and the output of `show version` junos command.

needs_info

**Comment by ansibot** [comment]
@alvarezluciano This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@alvarezluciano You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
